relay-general/src/protocol/transaction.rs

```diff
@@ -0,0 +1,157 @@
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "custom" => Ok(Self::Custom),
+            "url" => Ok(Self::Url),
+            "route" => Ok(Self::Route),
+            "view" => Ok(Self::View),
+            "component" => Ok(Self::Component),
+            "task" => Ok(Self::Task),
+            "unknown" => Ok(Self::Unknown),
+            s => Ok(Self::Other(s.to_owned())),
+        }
+    }
+}
+
+impl fmt::Display for TransactionSource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Custom => write!(f, "custom"),
+            Self::Url => write!(f, "url"),
+            Self::Route => write!(f, "route"),
+            Self::View => write!(f, "view"),
+            Self::Component => write!(f, "component"),
+            Self::Task => write!(f, "task"),
+            Self::Unknown => write!(f, "unknown"),
+            Self::Other(s) => write!(f, "{}", s),
+        }
+    }
+}
+
+impl Default for TransactionSource {
+    fn default() -> Self {
+        Self::Unknown
+    }
+}
+
+impl Empty for TransactionSource {
+    #[inline]
+    fn is_empty(&self) -> bool {
+        matches!(self, Self::Unknown)
+    }
+}
+
+impl FromValue for TransactionSource {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self> {
+        match String::from_value(value) {
+            Annotated(Some(value), mut meta) => match value.parse() {
+                Ok(source) => Annotated(Some(source), meta),
+                Err(_) => {
+                    meta.add_error(ErrorKind::InvalidData);
+                    meta.set_original_value(Some(value));
+                    Annotated(None, meta)
+                }
+            },
+            Annotated(None, meta) => Annotated(None, meta),
+        }
+    }
+}
+
+impl IntoValue for TransactionSource {
+    fn into_value(self) -> Value
+    where
+        Self: Sized,
+    {
+        Value::String(self.to_string())
+    }
+
+    fn serialize_payload<S>(&self, s: S, _behavior: SkipSerialization) -> Result<S::Ok, S::Error>
+    where
+        Self: Sized,
+        S: serde::Serializer,
+    {
+        serde::Serialize::serialize(&self.to_string(), s)
+    }
+}
+
+impl ProcessValue for TransactionSource {}
+
+/// Additional information about the name of the transaction.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct TransactionInfo {
+    /// Describes how the name of the transaction was determined.
+    ///
+    /// This will be used by the server to decide whether or not to scrub identifiers from the
+    /// transaction name, or replace the entire name with a placeholder.
+    pub source: Annotated<TransactionSource>,
+
+    /// The unmodified transaction name as obtained by the source.
+    ///
+    /// This value will only be set if the transaction name was modified during event processing.
```

Comment:
actually, don't we have _meta for original data?

Reply:
In the DACI it was decided that we do not use _meta to make the information more accessible to the product.